compiler/java/src/test/java/com/c/filereader/FileReaderTest.java

```diff
@@ -0,0 +1,460 @@
+package com.c.filereader;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import static com.c.testfiles.FileToString.fileToString;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class FileReaderTest {
+
+    // ---------------------------------------- EOF  -------------------------------------------------------------------
+
+    /**
+     * Checks that EOF is reached.
+     */
+    private void expectEOF(FileReader reader, int maxLookAhead) throws UnexpectedCharError {
+        assertThat(reader.lookAhead()).isEqualTo((char) -1);
+        reader.expect((char) -1);
+
+        Character[] bufferExpect = new Character[maxLookAhead];
+        Arrays.fill(bufferExpect, (char) -1);
+        assertThat(reader.lookAheads()).isEqualTo(bufferExpect);
+    }
+
+    @Test
+    @DisplayName("EOF in empty file")
+    public void testEmptyFile() throws UnexpectedCharError, MaxLookAheadError {
+        // Initialize
+        int maxLookAhead = 4;
+        String file = "";
+        FileReader reader = new FileReader(file, maxLookAhead);
+
+        expectEOF(reader, maxLookAhead);
+
+        // Advance and still be at EOF
+        reader.advance();
```

Comment:
for loop?

Reply:
Maybe, but I find it more explicit for these tests to see the reader advancing with every line. Can I leave it like it is?

Reply:
Sure